ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/common/RecoveryEnvironmentBean.java

```diff
@@ -597,4 +599,29 @@ public void setTimeoutSocket(boolean timeoutSocket)
     {
         this.timeoutSocket = timeoutSocket;
     }
+
+    /**
+     * This method gives information about the behaviour of
+     * {@link com.arjuna.ats.arjuna.recovery.RecoveryManager} when suspending.
+     * 
+     * @return true if {@link com.arjuna.ats.arjuna.recovery.RecoveryManager} should wait that all
+     * {@link RecoveryModule} implementations recover all their transactions before shutting down;
+     * false otherwise.
+     */
+    public boolean isWaitForRecovery()
+    {
+        return waitForRecovery;
+    }
+
+    /**
+     * Configure the suspension of {@link com.arjuna.ats.arjuna.recovery.RecoveryManager}
+     * 
+     * @param waitForRecovery true if {@link com.arjuna.ats.arjuna.recovery.RecoveryManager} should
+     * wait that all {@link RecoveryModule} implementations recover all their transactions before
```

Comment:
I am wondering about the word `all` (here and the isWaitForRecover). Maybe it is `all participating`. I say this because the default implementation does not wait to recover the module and maybe some implementations (e.g. provided by the user might not provide an implementation).

Reply:
Good point. It used to be `...all {@link SuspendBlockingRecoveryModule} implementations}` (which made more sense). What about `...all {@link RecoveryModule} implementing {@link RecoveryModule#hasWork()}`?

Reply:
I am not sure, because they all _do_ implement it because there is a default implementation. Maybe overriding implementations?

Reply:
Overriding implementations sounds good :-) Thanks Tom